frontend/src/views/Profile/Profile.tsx

```diff
@@ -0,0 +1,103 @@
+import styles from "./Profile.module.scss";
+import { useParams } from "react-router-dom";
+import React, { useState, useEffect } from "react";
+import { ProfileInterface } from "../../components/types/ProfileInterface";
+import { AuctionWithProductInfo } from "../../components/types/AuctionWithProductInfo";
+import { ProfileInfoPanel } from "../../components";
+import imageIcon from "../../assets/imageIcon.svg";
+import editIcon from "../../assets/editIcon.svg";
+
+const profileDisplayOptions = [
+    { value: "onsale", label: "On Sale" },
+    { value: "collectibles", label: "Collectibles" },
+    { value: "created", label: "Created" },
+    { value: "likes", label: "Likes" },
+    { value: "following", label: "Following" },
+    { value: "followers", label: "Followers" },
+];
+
+async function getProfile(profileID: string): Promise<ProfileInterface | null> {
+    try {
+        const profileInfo = await fetch(`http://localhost:8000/profiles/${profileID}`);
+        return profileInfo.json();
+    } catch (err) {
+        return null;
+    }
+}
+
+async function getUserAuctions(profileID: string): Promise<AuctionWithProductInfo[] | null> {
+    try {
+        const auctionsResponse = await fetch(
+            `http://localhost:8000/auctions/?full=true&profileID=${profileID}`
+        );
+        return auctionsResponse.json();
+    } catch (err) {
+        return null;
+    }
+}
+
+export function Profile() {
+    const { profileID } = useParams();
+    const [profile, setProfile] = useState<ProfileInterface | null>(null);
+    const [auctions, setAuctions] = useState<AuctionWithProductInfo[] | null>(null);
+    const [selectedProfileDisplay, setSelectedProfileDisplay] = useState<string>("onsale");
+
+    useEffect(() => {
+        (async () => {
+            if (!profileID) return;
+            setProfile(await getProfile(profileID));
+            setAuctions(await getUserAuctions(profileID));
+        })();
+    }, [profileID]);
+
+    console.log(auctions);
+    console.log(profile);
+
+    return (
+        <div className={styles.profileContainer}>
+            <div className={styles.coverPhotoWrapper}>
+                <img className={styles.coverPhoto} alt="Cover" src={profile?.coverPicture} />
+            </div>
+            <div className={styles.contentContainer}>
+                {profile && <ProfileInfoPanel profile={profile} />}
+                <div className={styles.mainContent}>
+                    <div className={styles.settingsButtons}>
+                        <button type="button" className={styles.buttonOnCoverPhoto}>
+                            Edit cover photo
+                            <img
```

Comment:
tip: if images/icons have no meaning and if there is a label or title for a button you can hide them on screen readers with `aria-hidden`